From d7d6452c5203ffd4de5b49d9e59765e9178beff4 Mon Sep 17 00:00:00 2001
From: Amit Pandit <quic_pamit@quicinc.com>
Date: Tue, 9 Jan 2024 16:40:38 +0530
Subject: [PATCH] wifi: ath12k: fix max num peer for peer create

max number of peer is set to 145, when we have a
max vdev setup i.e. 16 vdevs, the expected number
of peers would be 128.

Here in the issue case we are hitting Firmware assert,
when we try to create the 129th peer, as part of peer
create the check is not valid which allows the creation
of 129th peer.

Updating the code to ensure only 128 peers are allowed.

Signed-off-by: Amit Pandit <quic_pamit@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/peer.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/wireless/ath/ath12k/peer.c b/drivers/net/wireless/ath/ath12k/peer.c
index 5ba6e87..fc17fbf 100644
--- a/drivers/net/wireless/ath/ath12k/peer.c
+++ b/drivers/net/wireless/ath/ath12k/peer.c
@@ -612,7 +612,7 @@ int ath12k_peer_create(struct ath12k *ar, struct ath12k_link_vif *arvif,
 
 	lockdep_assert_held(&ar->conf_mutex);
 
-	if (ar->num_peers > (ar->max_num_peers - 1)) {
+	if (ar->num_peers >= (ar->max_num_peers - 1)) {
 		ath12k_warn(ar->ab,
 			    "failed to create peer due to insufficient peer entry resource in firmware\n");
 		return -ENOBUFS;
-- 
2.17.1

